commons/zenoh-protocol/src/proto/msg_writer.rs

```diff
@@ -521,7 +516,7 @@ impl MessageWriter for WBuf {
         zcheck!(self.write_string(&query.value_selector));
         zcheck!(self.write_zint(query.qid));
         if let Some(t) = query.target.as_ref() {
```

Comment:
if query.target != BestMatching instead? Here, if `query.target == Some(BestMatching)`, we're gonna spend a byte just to send it

Reply:
don't forget to change the `impl Header for Query` to match, and the reader side

Reply:
As said in the previous comment, I wouldn't change it now. In the forthcoming version of the protocol default values will be treated as such and not as options.